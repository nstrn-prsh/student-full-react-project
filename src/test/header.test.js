import "@testing-library/jest-dom/extend-expect";
import { cleanup, render } from "@testing-library/react";
import Header from "./Header";

afterEach(cleanup);

it("renderSnapshotHeader", () => {
     const { asFragment } = render(<Header text='hello' />);
     expect(asFragment()).toMatchSnapshot();
});

it("renders", () => {
     const { getByTestId, getByText } = render(<Header text='hello' />);
     expect(getByTestId("h1tag")).toHaveTextContent("hello");
     expect(getByText("hello")).toHaveClass("h1-class");
});

it("render button", () => {
     const { getByTestId } = render(<Header text='hello' />);
     expect(getByTestId("ok-button")).toHaveAttribute("disabled");
     expect(getByTestId("ok-button")).toHaveAttribute("type", "submit");
});
