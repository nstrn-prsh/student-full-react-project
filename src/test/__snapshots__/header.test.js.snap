// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderSnapshotHeader 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="h1-class"
      data-testid="h1tag"
    >
      hello
    </h1>
    <button
      data-testid="ok-button"
      disabled=""
      type="submit"
    >
      ok
    </button>
  </header>
</DocumentFragment>
`;
